ikarus/finiteelements/mechanics/truss.hh

```diff
@@ -0,0 +1,231 @@
+  KinematicVariables<ST> computeStrain(
+      const Requirement& par,
+      const std::optional<std::reference_wrapper<const Eigen::VectorX<ST>>>& dx = std::nullopt) const {
+    KinematicVariables<ST> kin;
+    const auto& localView    = underlying().localView();
+    const auto& tree         = localView.tree();
+    const auto numberOfNodes = tree.child(0).finiteElement().size();
+    auto& ele                = localView.element();
+    const auto X1            = Dune::toEigen(ele.geometry().corner(0));
+    const auto X2            = Dune::toEigen(ele.geometry().corner(1));
+    const auto u             = Dune::viewAsEigenMatrixAsDynFixed(displacement(par, dx)).transpose().eval();
+    const auto A1            = X2 - X1;
+    const auto ud1           = u.col(1) - u.col(0);
+
+    const Eigen::Vector<ST, worldDim> x1 = X1 + u.col(0);
+    const Eigen::Vector<ST, worldDim> x2 = X2 + u.col(1);
+
+    const double Lsquared = A1.squaredNorm();
+    const ST lsquared     = (x2 - x1).squaredNorm();
+
+    kin.L = sqrt(Lsquared);
+    kin.l = sqrt(lsquared);
+
+    // Green-Lagrange strains
+    kin.Egl = 0.5 * (lsquared - Lsquared) / Lsquared;
+
+    // First derivative of Egl w.r.t displacements
+    kin.dEdu.setZero(worldDim * numberOfNodes);
+    kin.dEdu << -A1 - ud1, A1 + ud1;
+    kin.dEdu /= Lsquared;
+
+    // Second derivative of Egl w.r.t displacements
+    kin.ddEddu.setIdentity(worldDim * numberOfNodes, worldDim * numberOfNodes);
+    kin.ddEddu.template topRightCorner<worldDim, worldDim>()   = -Eigen::Matrix<ST, worldDim, worldDim>::Identity();
+    kin.ddEddu.template bottomLeftCorner<worldDim, worldDim>() = -Eigen::Matrix<ST, worldDim, worldDim>::Identity();
+    kin.ddEddu /= Lsquared;
+
+    return kin;
+  }
+
+  using SupportedResultTypes = std::tuple<decltype(makeRT<ResultTypes::axialForce>())>;
+
+private:
+  template <template <typename, int, int> class RT>
+  static consteval bool canProvideResultType() {
+    return isSameResultType<RT, ResultTypes::axialForce>;
+  }
+
+public:
+  /**
+   * \brief Calculates a requested result at a specific local position.
+   *
+   * \param req The Requirement object holding the global solution.
+   * \param local Local position vector.
+   * \tparam RT The requested result type
+   * \return calculated result
+   *
+   * \tparam RT The type representing the requested result.
+   */
+  template <template <typename, int, int> class RT>
+  requires(canProvideResultType<RT>())
+  auto calculateAtImpl(const Requirement& req, [[maybe_unused]] const Dune::FieldVector<double, Traits::mydim>& local,
+                       Dune::PriorityTag<0>) const {
+    using RTWrapper = ResultWrapper<RT<double, myDim, myDim>, ResultShape::Vector>;
+    if constexpr (isSameResultType<RT, ResultTypes::axialForce>) {
+      const auto [L, l, Egl, dEdu, ddEddu] = computeStrain(req);
+      auto N = Eigen::Vector<double, 1>{E * A * Egl * l / L}; // Axial force in deformed configuration
```

Comment:
This force you compute here is the Cauchy axial force. Maybe also add PK2 axial force or maybe we want to rename axial force or do you think this is sufficient?

Reply:
Yes, I can add both `cauchyAxialForce` and `PK2AxialForce`. May be in the future this solution can be generalized where depending on the strain measure provided by the user, the `result` is determined, or something